quartz/components/scripts/search.inline.ts

```diff
@@ -248,6 +288,42 @@ document.addEventListener("nav", async (e: unknown) => {
     }
   }
 
+  async function fetchContent(slug: FullSlug): Promise<Element[] | undefined> {
+    if (fetchContentCache[slug]) {
+      return fetchContentCache[slug]
+    }
+
+    const targetUrl = resolveUrl(slug).toString()
+    const contents = await fetch(targetUrl)
+      .then((res) => res.text())
+      .then((contents) => {
+        if (contents === undefined) {
+          throw new Error(`Could not fetch ${targetUrl}`)
+        }
+        const html = p.parseFromString(contents ?? "", "text/html")
+        normalizeRelativeURLs(html, targetUrl)
+        return [...html.getElementsByClassName("popover-hint")]
+      })
+      .catch((err) => {
+        console.error(err)
+        throw new Error(`Could not fetch ${targetUrl}`)
+      })
```

Comment:
let's not catch and let promise reject instead of returning undefined

Reply:
done.